### ISSUE TITLE
Random Crashes

### ISSUE THREAD

**Issue by Voklen** [grkek/haki#4]
### Layout Version
commit 8df4f2bcbb9e0638c1024d8155806dcedd6c52ec

### System information
Arch KDE Plasma 5.22.5, Linux kernel 5.14.3

### Issue description
When running the default example program, it spontaneously crashes, sometimes it will work and sometimes it won't. When it crashes the error message it gives is either:
```
Building components...
Finished: 14.17ms
"T"
"Th"
"Thi"
"This"
"This "
"This i"
"This is"
"This is "
"This is a"
"This is an"
"This is an "
"This is an e"
"This is an ex"
"This is an exa"
"This is an exam"
"This is an examp"
"This is an exampl"
"This is an example"
Invalid memory access (signal 11) at address 0x55ba65bfca00
[0x55ba652d1026] *Exception::CallStack::print_backtrace:(Int32 | Nil) +118
[0x55ba652b364a] ~procProc(Int32, Pointer(LibC::SiginfoT), Pointer(Void), Nil) +330
[0x7f6b6f18d870] ???
[0x55ba65bfca00] ???
```
(at some random address)
or
```
Building components...
Finished: 14.28ms
"T"
"Th"
"Thi"
"This"
"This "
"This i"
"This is"
"This is "
"This is a"
"This is an"
"This is an "
"This is an e"
"This is an ex"
"This is an exa"
"This is an exam"
Invalid memory access (signal 11) at address 0x0
[0x55c2ce9ed026] *Exception::CallStack::print_backtrace:(Int32 | Nil) +118
[0x55c2ce9cf64a] ~procProc(Int32, Pointer(LibC::SiginfoT), Pointer(Void), Nil) +330
[0x7f536129a870] ???
[0x0] ???
```
(at address 0x0)

### Steps to reproduce
Follow the README.md instruction and run the default program. Then just click some buttons and type some text and it should crash.

**Comment by grkek** [comment]
- Notice: This project is experimental.

None of the shards I am using in this project are maintained by me, I don't know if the compiled Duktape or GTK is working properly, might even be an issue on my end!

**Comment by Voklen** [comment]
@grkek Yep, just reporting this in case it can give you some information to track down a bug. :+1:

**Comment by grkek** [comment]
I have seen this error before, not quite sure what causes it!

**Comment by grkek** [comment]
This issue is fixed by #6
